hummingbot/market/dolomite/dolomite_market

```diff
@@ -0,0 +1,1362 @@
+    OrderCancelledEvent,
+    MarketOrderFailureEvent,
+    BuyOrderCreatedEvent,
+    SellOrderCreatedEvent,
+    TradeType,
+    OrderType,
+    TradeFee,
+)
+from hummingbot.core.data_type.limit_order import LimitOrder
+from hummingbot.core.data_type.order_book cimport OrderBook
+from hummingbot.core.data_type.order_book_tracker import OrderBookTrackerDataSourceType
+from hummingbot.market.market_base cimport MarketBase
+from hummingbot.market.ddex.ddex_order_book_tracker import DolomiteOrderBookTracker #
+from hummingbot.core.network_iterator import NetworkStatus
+from hummingbot.wallet.ethereum.web3_wallet import Web3Wallet
+
+import binascii 
+import time 
+import json
+
+
+s_logger = None
+s_decimal_0 = Decimal(0)
+NaN = float("nan")
+
+
+cdef class DolomiteMarketTransactionTracker(TransactionTracker):
+    cdef:
+        DolomiteMarket _owner
+
+    def __init__(self, owner: DolomiteMarket):
+        super().__init__()
+        self._owner = owner
+
+    cdef c_did_timeout_tx(self, str tx_id):
+        TransactionTracker.c_did_timeout_tx(self, tx_id)
+        self._owner.c_did_timeout_tx(tx_id)
+
+
+cdef class TradingRule:
+    cdef:
+        public str symbol
+        public double min_maker_order_size
+        public double min_taker_order_size
+        public int price_decimals               # max amount of decimals in a price
+        public int amount_decimals              # max amount of decimals in an amount
+        public bint supports_limit_orders       # if limit order is allowed for this trading pair
+        public bint supports_market_orders      # if market order is allowed for this trading pair
+
+    @classmethod
+    def parse_exchange_info(cls, markets: List[Dict[str, Any]], min_maker_order_usd, min_taker_order_usd, weth_price) -> List[TradingRule]:
+        cdef:
+            list retval = []
+        
+        for market in markets:
+            try:
+                
+                symbol = market["market"]
+                
+                minMakerOrderSize = None
+                minTakerOrderSize = None
+                
+                #Convert min maker order / min taker order size to quantity of base token
+                
+                if symbol == "BAT-WETH": #quote token ETH
+                    min_maker_order = min_maker_order_usd / weth_price
+                    min_taker_order = min_taker_order_usd / weth_price
+                    
+                    minMakerOrderSize = min_maker_order / (float(market["current_price"]["amount"]) / math.pow(10, 18))
+                    minTakerOrderSize = min_taker_order / (float(market["current_price"]["amount"]) / math.pow(10, 18))
+                    
+                else: #quote token DAI
+                
+                    minMakerOrderSize = min_maker_order_usd / (float(market["current_price"]["amount"]) / math.pow(10, 18))
+                    minTakerOrderSize = min_taker_order_usd / (float(market["current_price"]["amount"]) / math.pow(10, 18))
```

Comment:
It's ok to hardcode 18 here because we know the precision of WETH and DAI are 18, but be careful of this in the future